cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -582,6 +583,29 @@ func (m *McmManager) GetMachineDeploymentNodes(machinedeployment *MachineDeploym
 	return nodes, nil
 }
 
+// validateNodeTemplate function validates the NodeTemplate object of the MachineClass
+func validateNodeTemplate(nodeTemplateAttributes *v1alpha1.NodeTemplate) error {
+	var allErrs []error
+
+	capacityAttributes := []v1.ResourceName{"cpu", "gpu", "memory"}
+
+	for _, attribute := range capacityAttributes {
+		if _, ok := nodeTemplateAttributes.Capacity[attribute]; !ok {
+			errMessage := fmt.Errorf("CPU, GPU and memory fields are mandatory")
+			klog.Warning(errMessage)
+			allErrs = append(allErrs, errMessage)
```

Comment:
we are printing the same error message more than once , so in case lets say cpu, gpu are missing same error message will be printed twice and appended to allErrs twice.

Reply:
also warning message will repeat

Reply:
Yeah, this makes sense. A `break` statement should do the trick.